### PR TITLE
Added CPP solution for usaco-940

### PR DESCRIPTION
**Pull request by LordLava**
_If any of the below doesn't apply to this Pull Request, mark the checkbox as done._

- [x] I have tested my code.
- [x] I have added my solution according to the steps [here](https://usaco.guide/general/adding-solution#steps).
- [x] I have followed the code conventions mentioned [here](https://usaco.guide/general/adding-solution/#code-conventions), which include the following: 
  - I understand that if it is clear that I have not attempted to follow these guidelines (ex. if I have not used tabs to indent), my PR will be closed.
  - If changes are requested, I will re-request the review after making them.

### CODE REVIEW

File: solutions/usaco-940.mdx [hunk 1]
```diff
@@ -87,6 +87,61 @@ public class Factory {
 ```
 
 </JavaSection>
+<CPPSection>
+
+```cpp
+#include <bits/stdc++.h>
+
```
Comment (ryanchou-dev) [inline solutions/usaco-940.mdx:94]:
```suggestion
```

File: solutions/usaco-940.mdx [hunk 2]
```diff
@@ -87,6 +87,61 @@ public class Factory {
...
+using namespace std;
+
+void setIO(string name=""){ // FastIO See General -> Input/Output
+    ios_base::sync_with_stdio(0); cin.tie(0);
+    if((int)name.size() > 0){
+        freopen((name+".in").c_str(), "r", stdin);
+        freopen((name+".out").c_str(), "w", stdout);
+    }
+}
```
Comment (ryanchou-dev) [inline solutions/usaco-940.mdx:103]: recommend detemplifying

File: solutions/usaco-940.mdx [hunk 3]
```diff
@@ -87,6 +87,61 @@ public class Factory {
...
+
+int main(){
+    setIO("factory");
+    int N;
+    cin >> N;
+    int outgoing[N] = {0};
+    // N - 1 walkways
+    for(int i = 0; i < N -1; i++){
+        int a,b;
```
Comment (ryanchou-dev) [inline solutions/usaco-940.mdx:112]: spaces after control keywords and operators
```
for (int i = 0; i < n; i++) {
    int a, b;
}
```

File: solutions/usaco-940.mdx [hunk 4]
```diff
@@ -87,6 +87,61 @@ public class Factory {
...
+        outgoing[a-1]++;
+
+    }
+    int ans = -1;
+    for (int i = 0; i < N; i++){
+        // only stations with outdegree 0 could be the sink
+        if(outgoing[i] == 0){
+            if(ans == -1){
+                // found the sink
+                ans = i + 1; // Array index begins with 0
```
Comment (ryanchou-dev) [inline solutions/usaco-940.mdx:125]:
```suggestion
                ans = i + 1;  // array index begins with 0
```
keep casing consistent & two spaces after line for inline comments

File: solutions/usaco-940.mdx [hunk 5]
```diff
@@ -87,6 +87,61 @@ public class Factory {
...
+            }
+            else{
+                // found two sinks
+                // so it is impossible to get from one to another
```
Comment (ryanchou-dev) [inline solutions/usaco-940.mdx:129]: should be ok to combine, if over 80 characters, multiline comments like:
```
/*
 * one
 * two
 * three
 */
```

File: solutions/usaco-940.mdx [hunk 6]
```diff
@@ -87,6 +87,61 @@ public class Factory {
...
+                ans = -1;
+                break;
+            }
+        }
+    }
+
+    cout << ans << flush;
```
Comment (ryanchou-dev) [inline solutions/usaco-940.mdx:136]:
```suggestion
    cout << ans << endl;
```

File: solutions/usaco-940.mdx [hunk 7]
```diff
@@ -87,6 +87,61 @@ public class Factory {
...
+        freopen((name+".in").c_str(), "r", stdin);
+        freopen((name+".out").c_str(), "w", stdout);
+    }
+}
+
+int main(){
+    setIO("factory");
+    int N;
+    cin >> N;
+    int outgoing[N] = {0};
```
Comment (ryanchou-dev) [inline solutions/usaco-940.mdx:109]: arrays are not meant to be dynamically sized, use vectors instead

File: solutions/usaco-940.mdx [hunk 8]
```diff
@@ -87,6 +87,61 @@ public class Factory {
...
+    // N - 1 walkways
+    for(int i = 0; i < N -1; i++){
+        int a,b;
+        cin >> a >> b;
+        // count the outdegree of this station
+        // Array index begins with 0, station begins with 1, so -1
+        outgoing[a-1]++;
+
+    }
+    int ans = -1;
```
Comment (ryanchou-dev) [inline solutions/usaco-940.mdx:119]:
```suggestion

    int ans = -1;
```

**Review by ryanchou-dev — CHANGES_REQUESTED** [review]
reflect suggestions for the rest of the code :D

happy first contribution 🎉

File: solutions/usaco-940.mdx (hunk 3 above)
Comment (LordLava) [reply solutions/usaco-940.mdx]: Done

File: solutions/usaco-940.mdx (hunk 5 above)
Comment (LordLava) [reply solutions/usaco-940.mdx]: Done

File: solutions/usaco-940.mdx (hunk 7 above)
Comment (LordLava) [reply solutions/usaco-940.mdx]: Done

File: solutions/usaco-940.mdx [hunk 9]
```diff
@@ -87,6 +87,54 @@ public class Factory {
...
+#include <bits/stdc++.h>
+using namespace std;
+
+int main() {
+    freopen("factory.in", "r", stdin);
+    freopen("factory.out", "w", stdout);
+
+    int N;
+    cin >> N;
+    vector<int> outgoing(N,0);
```
Comment (ryanchou-dev) [inline solutions/usaco-940.mdx:102]: ^

File: solutions/usaco-940.mdx [hunk 10]
```diff
@@ -87,6 +87,54 @@ public class Factory {
...
+    // N - 1 walkways
+    for (int i = 0; i < N -1; i++) {
```
Comment (ryanchou-dev) [inline solutions/usaco-940.mdx:104]: ^

File: solutions/usaco-940.mdx [hunk 11]
```diff
@@ -87,6 +87,54 @@ public class Factory {
...
+        int a, b;
+        cin >> a >> b;
+        // count the outdegree of this station
+        // Array index begins with 0, station begins with 1, so -1
+        outgoing[a-1]++;
```
Comment (ryanchou-dev) [inline solutions/usaco-940.mdx:109]: ^

**Review by ryanchou-dev — CHANGES_REQUESTED** [review]

File: solutions/usaco-940.mdx [hunk 12]
```diff
@@ -87,6 +87,54 @@ public class Factory {
...
+
+    int N;
+    cin >> N;
+    vector<int> outgoing(N, 0);
+    // N - 1 walkways
+    for (int i = 0; i < N - 1; i++) {
+        int a, b;
+        cin >> a >> b;
+        // count the outdegree of this station
+        // Array index begins with 0, station begins with 1, so -1
```
Comment (ryanchou-dev) [inline solutions/usaco-940.mdx:108]: ^

File: solutions/usaco-940.mdx [hunk 13]
```diff
@@ -87,6 +87,54 @@ public class Factory {
...
+
+    int ans = -1;
+    for (int i = 0; i < N; i++) {
+        // only stations with outdegree 0 could be the sink
+        if (outgoing[i] == 0) {
+            if (ans == -1) {
+                // found the sink
+                ans = i + 1;  // array index begins with 0
+            }
+            else {
```
Comment (ryanchou-dev) [inline solutions/usaco-940.mdx:121]:
```suggestion
            } else {
```

**Review by ryanchou-dev — CHANGES_REQUESTED** [review]
indent with tabs

**Review by ryanchou-dev — APPROVED** [review]

File: solutions/usaco-940.mdx [hunk 14]
```diff
@@ -85,10 +85,52 @@ public class Factory {
...
+#include <bits/stdc++.h>
+using namespace std;
+
+int main() {
+	freopen("factory.in", "r", stdin);
+	freopen("factory.out", "w", stdout);
+
+	int N;
+	cin >> N;
+	vector<int> outgoing(N, 0);
```
Comment (envyaims) [inline solutions/usaco-940.mdx:101]:
```suggestion
	vector<int> outgoing(N);
```

**Review by envyaims — APPROVED** [review]

**Pull request merged**